kv/index.go

```diff
@@ -195,16 +197,20 @@ func (i *Index) Walk(ctx context.Context, tx Tx, foreignKey []byte, visitFn Visi
 		return err
 	}
 
-	return indexWalk(ctx, cursor, sourceBucket, visitFn)
+	return indexWalk(ctx, foreignKey, cursor, sourceBucket, visitFn)
 }
 
 // indexWalk consumes the indexKey and primaryKey pairs in the index bucket and looks up their
 // associated primaryKey's value in the provided source bucket.
 // When an item is located in the source, the provided visit function is called with primary key and associated value.
-func indexWalk(ctx context.Context, indexCursor ForwardCursor, sourceBucket Bucket, visit VisitFunc) (err error) {
+func indexWalk(ctx context.Context, foreignKey []byte, indexCursor ForwardCursor, sourceBucket Bucket, visit VisitFunc) (err error) {
 	var keys [][]byte
 	for ik, pk := indexCursor.Next(); ik != nil; ik, pk = indexCursor.Next() {
-		keys = append(keys, pk)
+		if fk, _, err := indexKeyParts(ik); err != nil {
```

Comment:
🙌 awesome

Reply:
Actually @stuartcarnie you know what I've just thought of something better.
If we curse `foreignKey + "/"` then we push down the actual prefix to the db.
This does make me think the foreign key needs to be escaped for `"/"` as well though.

Reply:
It is unclear what I should change here – should we add some code to validate that foreign keys do not have `/` for now?

Reply:
Spoke with @GeorgeMac and I am going to reject foreign keys with a `/`. Clients of the `Index` will be expected to partition their foreign keys with characters other than a `/`